### PR TITLE
Gjoranv/create registry from config

### PR DESCRIPTION
**Pull request by gjoranv**
@bratseth, please review the new constructor code carefully.

Next, we can have this injected to SearchHandler instead of creating a new registry there.

### CODE REVIEW

**Review by baldersheim — CHANGES_REQUESTED** [review]
Looks good to me. I would just include the change in SearchHandler to avoid blink-dpa being the first one using it.

**Review by baldersheim — APPROVED** [review]

**Comment by gjoranv** [comment]
Done, but that was a _painful_ reminder about the integration style tests in container-search. These were written before the DI framework, but were adapted to it by converting the test files for configs that are long gone now (handlers and chains) into the new `components` config.

IMO, these tests should be rewritten to plain unit tests, and some should just be removed. No need to test DI here anymore.

**Review by baldersheim — APPROVED** [review]
LGTM. Sometimes pain is necessary go get going :)

File: container-search/src/main/java/com/yahoo/search/query/profile/compiled/CompiledQueryProfileRegistry.java [hunk 1]
```diff
@@ -18,6 +23,15 @@
     
     private final QueryProfileTypeRegistry typeRegistry;
 
+    @Inject
+    public CompiledQueryProfileRegistry(QueryProfilesConfig config) {
+        QueryProfileRegistry registry = QueryProfileConfigurer.createFromConfig(config);
+        typeRegistry = registry.getTypeRegistry();
+        for (QueryProfile inputProfile : registry.allComponents()) {
```
Comment (bratseth) [inline container-search/src/main/java/com/yahoo/search/query/profile/compiled/CompiledQueryProfileRegistry.java:30]: Why not just do 
QueryProfileCompiler.compile(registry)
here?

**Review by bratseth — APPROVED** [review]
LGTM but what is it that you want to achieve, recompiling less?

File: container-search/src/main/java/com/yahoo/search/query/profile/compiled/CompiledQueryProfileRegistry.java (hunk 1 above)
Comment (gjoranv) [reply container-search/src/main/java/com/yahoo/search/query/profile/compiled/CompiledQueryProfileRegistry.java:30]: I didn't want to create a temporary object of the same type as the class itself. There is room for improvement, but it requires creating even more public methods and/or constructors.

**Pull request merged**